config_src/external/OCEAN_stochastic_phyiscs/MOM_stochastics.F90

```diff
@@ -0,0 +1,157 @@
+use MOM_domains,             only : root_PE,num_PEs
+use MOM_coms,                only : Get_PElist
+use stochastic_physics,      only : init_stochastic_physics_ocn, run_stochastic_physics_ocn
+use get_stochy_pattern_mod, only: write_stoch_restart_ocn
+
+#include <MOM_memory.h>
+
+implicit none ; private
+
+public stochastics_init, update_stochastics, write_mom_restart_stoch
+
+!> This control structure holds parameters for the MOM_stochastics module
+type, public:: stochastic_CS
+  logical :: do_sppt                 !< If true, stochastically perturb the diabatic
+  logical :: pert_epbl       !! If true, then randomly perturb the KE dissipation and genration terms
+  integer :: id_sppt_wts  = -1
+  integer :: id_epbl1_wts=-1,id_epbl2_wts=-1
+  ! stochastic patterns
+  real, allocatable :: sppt_wts(:,:)  !< Random pattern for ocean SPPT
+                                     !! tendencies with a number between 0 and 2
+  real, allocatable :: epbl1_wts(:,:) !< Random pattern for K.E. generation
+  real, allocatable :: epbl2_wts(:,:) !< Random pattern for K.E. dissipation
+  type(diag_ctrl), pointer :: diag   !< structure used to regulate timing of diagnostic output
+  type(time_type), pointer :: Time !< Pointer to model time (needed for sponges)
+end type stochastic_CS
+
+contains
+
+!!   This subroutine initializes the stochastics physics control structure.
+subroutine stochastics_init(dt, grid, GV, CS, param_file, diag, Time)
+  real, intent(in)                     :: dt       !< time step [T ~> s]
+  type(ocean_grid_type),   intent(in)  :: grid     ! horizontal grid information
+  type(verticalGrid_type), intent(in)  :: GV       ! vertical grid structure
+  type(stochastic_CS), pointer,     intent(inout):: CS
+  type(param_file_type),   intent(in)    :: param_file !< A structure to parse for run-time parameters
+  type(diag_ctrl), target, intent(inout) :: diag             !< structure to regulate diagnostic output
+  type(time_type), target                :: Time             !< model time
+  ! Local variables
+  integer,allocatable :: pelist(:) ! list of pes for this instance of the ocean
+  integer :: mom_comm          ! list of pes for this instance of the ocean
+  integer :: num_procs         ! number of processors to pass to stochastic physics
+  integer :: iret              ! return code from stochastic physics
+  integer :: me                !  my pe
+  integer :: master            !  root pe
+  integer :: nx                ! number of x-points including halo
+  integer :: ny                ! number of x-points including halo
+
+! This include declares and sets the variable "version".
+#include "version_variable.h"
+  character(len=40)  :: mdl = "ocean_stochastics_init"  ! This module's name.
+
+  call callTree_enter("ocean_model_stochastic_init(), MOM_stochastics.F90")
+  if (associated(CS)) then
+    call MOM_error(WARNING, "MOM_stochastics_init called with an "// &
+                            "associated control structure.")
```

Comment:
I am confused by this "warning", should this instead be: "... UNassociated control structure." ? or else, "ALREADY associated control structure." ?

Reply:
Phil has modified this